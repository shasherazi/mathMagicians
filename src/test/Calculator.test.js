import React from 'react';
import renderer from 'react-test-renderer';
import { render, waitFor } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import Calculator from '../components/Calculator';

it('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('The user types 4 + 4', () => {
  const tree = render(<Calculator />);
  const calculatorInput = tree.container.querySelector('.calculator-display');
  let button = tree.getByText('4');
  fireEvent.click(button);
  waitFor(() => expect(calculatorInput.innerHTML).toBe('4'));
  button = tree.getByText('+');
  fireEvent.click(button);
  waitFor(() => expect(calculatorInput.innerHTML).toBe('4 +'));
  button = tree.getByText('4');
  fireEvent.click(button);
  waitFor(() => expect(calculatorInput.innerHTML).toBe('4 + 4'));
  button = tree.getByText('=');
  fireEvent.click(button);
  waitFor(() => expect(calculatorInput.innerHTML).toBe('8'));
});

test('The user tries to divide by zero', () => {
  const tree = render(<Calculator />);
  const calculatorInput = tree.container.querySelector('.calculator-display');
  let button = tree.getByText('4');
  fireEvent.click(button);
  waitFor(() => expect(calculatorInput.innerHTML).toBe('4'));
  button = tree.getByText('รท');
  fireEvent.click(button);
  waitFor(() => expect(calculatorInput.innerHTML).toBe('4 รท'));
  button = tree.getByText('0');
  fireEvent.click(button);
  waitFor(() => expect(calculatorInput.textContent).toBe('4 รท 0'));
  button = tree.getByText('=');
  fireEvent.click(button);
  waitFor(() => expect(calculatorInput.innerHTML).toBe('Cannot divide by zero'));
});
