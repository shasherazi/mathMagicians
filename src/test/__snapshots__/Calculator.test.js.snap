// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calculator"
>
  <div
    className="calculator-display"
  >
    0
  </div>
  <div
    className="calculator-buttons"
  >
    <button
      className="operation-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="operation-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="operation-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operation-btn calculator-button right"
      id={null}
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="number-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="number-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="number-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operation-btn calculator-button right"
      id={null}
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="number-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="number-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="number-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operation-btn calculator-button right"
      id={null}
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="number-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="number-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="number-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operation-btn calculator-button right"
      id={null}
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="number-btn calculator-button"
      id="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="operation-btn calculator-button"
      id={null}
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operation-btn calculator-button right"
      id={null}
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
